v2/internal/controllers/controller_resources.go

```diff
@@ -6,16 +6,33 @@ Licensed under the MIT license.
 package controllers
 
 import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	resources "github.com/Azure/azure-service-operator/v2/api/resources/v1alpha1api20200601"
+	. "github.com/Azure/azure-service-operator/v2/internal/logging"
+	"github.com/Azure/azure-service-operator/v2/internal/reflecthelpers"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/registration"
 )
 
-func GetKnownStorageTypes() []client.Object {
+func GetKnownStorageTypes() []registration.StorageTypeInfo {
 	knownTypes := getKnownStorageTypes()
 
-	knownTypes = append(knownTypes, new(resources.ResourceGroup))
+	knownTypes = append(
+		knownTypes,
+		registration.StorageTypeInfo{
+			Obj: new(resources.ResourceGroup),
```

Comment:
Would be better if we rename `Obj` to something which makes more sense.

Reply:
Agree that the `Obj` naming is pretty "generic", but it is what we call it in other places, such as the [generic controller](https://github.com/Azure/azure-service-operator/blob/main/v2/internal/controllers/generic_controller.go#L101). The generic controller took the abbreviation from controller-runtime, where it's [object](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/builder/controller.go#L81), which I don't think is tons clearer.

I've added some documentation onto the `StorageType` that describes what Obj is, which maybe will help clarify things, but left it named as is due to it matching parameters elsewhere.